aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -1902,6 +1920,110 @@ In the following `App` component example:
 > [!NOTE]
 > Not throwing if the cancellation token in <xref:Microsoft.AspNetCore.Components.Routing.NavigationContext> is canceled can result in unintended behavior, such as rendering a component from a previous navigation.
 
+## Handle location changes in a component
+
+`RegisterLocationChangingHandler` registers a handler to process incoming navigation events. The handler's context provided by `LocationChangingContext` includes the following properties:
+
+* `TargetLocation`: Gets the target location.
+* `HistoryEntryState`: Gets the state associated with the target history entry.
+* `IsNavigationIntercepted`: Gets whether the navigation was intercepted from a link.
+* `CancellationToken`: Gets a <xref:System.Threading.CancellationToken> to determine if the navigation was canceled, for example, to determine if the user triggered a different navigation.
+* `PreventNavigation`: Called to prevent the navigation from continuing.
+
+A component can register multiple location changing handlers, and calling <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A> invokes all of the registered handlers. Registered handlers are automatically disposed by the Navigation Manager when the component is disposed.
```

Comment:
> Registered handlers are automatically disposed by the Navigation Manager when the component is disposed.

This is only true when using the `<NavigationLock />` component. If a handler is created using `NavigationManager.RegisterLocationChangingHandler()`, then the handler must be manually disposed, e.g.:
```csharp
var registration = NavigationManager.RegisterLocationChangingHandler(OnLocationChanging);
// ...
registration.Dispose();
```